ako-gateway-api/k8s/validator.go

```diff
@@ -356,6 +441,18 @@ func validateParentReference(key string, httpRoute *gatewayv1.HTTPRoute, httpRou
 			utils.AviLog.Errorf("key: %s, msg: unable to find the listener from the Section Name %s in Parent Reference %s", key, name, listenerName)
 			err := fmt.Errorf("Invalid listener name provided")
 			defaultCondition.
+				Reason(string(gatewayv1.RouteReasonNoMatchingParent)).
+				Message(err.Error()).
+				SetIn(&httpRouteStatus.Parents[*parentRefIndexInHttpRouteStatus].Conditions)
+			*parentRefIndexInHttpRouteStatus = *parentRefIndexInHttpRouteStatus + 1
+			return err
+		}
+		if gateway.Status.Listeners[i].Conditions[0].Type == string(gatewayv1.GatewayConditionAccepted) && gateway.Status.Listeners[i].Conditions[0].Status == metav1.ConditionFalse {
```

Comment:
Are we fetching Patched `Gateway With status` only to check this condition? If yes, Is it possible to optimize that approach as we are doing return of patched gateway objects from Patch method?